packages/database/src/core/util/ServerValues.ts

```diff
@@ -67,13 +116,17 @@ export const resolveDeferredValue = function(
  */
 export const resolveDeferredValueTree = function(
   tree: SparseSnapshotTree,
+  syncTree: SyncTree,
   serverValues: Object
 ): SparseSnapshotTree {
   const resolvedTree = new SparseSnapshotTree();
   tree.forEachTree(new Path(''), function(path, node) {
+    const existing = syncTree.calcCompleteEventCache(path);
+    assert(existing != null && typeof existing != 'undefined',
```

Comment:
In Firestore we just do `assert(!!existing)`.

If you want to keep it more explicit, please change to `existing !== null`. `existing != null` already implicitly checks for undefined, but should be rejected by the linter (which we may not have set up correctly).

Reply:
Used !==